### ISSUE TITLE
Url rewrite should support option updates

### ISSUE THREAD

**Issue by dlyz** [aspnet/BasicMiddleware#381]
Use case: I have a static files host application with some complex configuration and it should support updating the redirects config on the fly.
Currently, RewriteMiddleware accepts IOptions and reads its value only once. It would be great to replace IOptions with IOptionsMonitor and use its CurrentValue for each request.
Similar feature: https://github.com/aspnet/BasicMiddleware/issues/317

**Comment by jkotalik** [comment]
I think this is a good ask for many of our middleware. @HaoK Is IOptionsMonitor primarily used for file based configuration changes?

**Comment by HaoK** [comment]
Yes the reload on change file based configuration changes come for free with IOptionsMonitor
